Test failure for sparse mode basis IO

This has likely to do with big endian coming from fits files.

Full test output:
```
______________________________ test_mode_basis_io ______________________________

    def test_mode_basis_io():
        grid = make_pupil_grid(128)
        mode_bases = [
            make_zernike_basis(20, 1, grid, 1),
            make_xinetics_influence_functions(grid, 8, 1 / 8)
        ]
    
        formats = ['asdf', 'fits', 'fits.gz', 'pkl', 'pickle']
        filenames = ['mode_basis_test.' + fmt for fmt in formats]
    
        for mode_basis in mode_bases:
            for fname in filenames:
                write_mode_basis(mode_basis, fname)
>               new_mode_basis = read_mode_basis(fname)

tests/test_util.py:52: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
hcipy/util/io.py:367: in read_mode_basis
    return ModeBasis.from_dict(tree)
hcipy/mode_basis/mode_basis.py:90: in from_dict
    return mode_basis.to_sparse()
hcipy/mode_basis/mode_basis.py:160: in to_sparse
    T = scipy.sparse.csc_matrix(self._transformation_matrix)
../../../miniconda3/envs/ci-env/lib/python3.10/site-packages/scipy/sparse/_compressed.py:94: in __init__
    coo = self._coo_container(arg1, dtype=dtype)
../../../miniconda3/envs/ci-env/lib/python3.10/site-packages/scipy/sparse/_coo.py:95: in __init__
    self.data = getdata(M[coords], copy=copy, dtype=dtype)
../../../miniconda3/envs/ci-env/lib/python3.10/site-packages/scipy/sparse/_sputils.py:1[50](https://github.com/ehpor/hcipy/actions/runs/15201313437/job/42772851289?pr=261#step:7:51): in getdata
    getdtype(data.dtype)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

dtype = dtype('>f8'), a = None, default = None

    def getdtype(dtype, a=None, default=None):
        """Form a supported numpy dtype based on input arguments.
    
        Returns a valid ``numpy.dtype`` from `dtype` if not None,
        or else ``a.dtype`` if possible, or else the given `default`
        if not None, or else raise a ``TypeError``.
    
        The resulting ``dtype`` must be in ``supported_dtypes``:
            bool_, int8, uint8, int16, uint16, int32, uint32,
            int64, uint64, longlong, ulonglong, float32, float64,
            longdouble, complex64, complex128, clongdouble
        """
        if dtype is None:
            try:
                newdtype = a.dtype
            except AttributeError as e:
                if default is not None:
                    newdtype = np.dtype(default)
                else:
                    raise TypeError("could not interpret data type") from e
        else:
            newdtype = np.dtype(dtype)
    
        if newdtype not in supported_dtypes:
            supported_dtypes_fmt = ", ".join(t.__name__ for t in supported_dtypes)
>           raise ValueError(f"scipy.sparse does not support dtype {newdtype.name}. "
                             f"The only supported types are: {supported_dtypes_fmt}.")
E           ValueError: scipy.sparse does not support dtype float64. The only supported types are: bool, int8, uint8, int16, uint16, int32, uint32, int64, uint64, longlong, ulonglong, float32, float64, longdouble, complex64, complex128, clongdouble.
```

--- a/hcipy/mode_basis/mode_basis.py
+++ b/hcipy/mode_basis/mode_basis.py
@@ -157,7 +157,15 @@ class ModeBasis(object):
             if self._transformation_matrix.ndim != 2:
                 raise TypeError('Cannot sparsify a mode basis of tensor fields')
 
-            T = scipy.sparse.csc_matrix(self._transformation_matrix)
+            # Ensure the transformation matrix is little endian.
+            # Scipy.sparse cannot deal with big endian sparse matrices.
+            if self._transformation_matrix.dtype.byteorder == '>':
+                T = self._transformation_matrix.byteswap()
+                T = T.view(T.dtype.newbyteorder('<'))
+            else:
+                T = self._transformation_matrix
+
+            T = scipy.sparse.csc_matrix(T)
             T.eliminate_zeros()
             return ModeBasis(T, self.grid)
 

